### ISSUE TITLE
ServiceInsight does not allow running multiple instances on the same machine via remote desktop

### ISSUE THREAD

**Issue by kbaley** [Particular/ServiceInsight#1033]
## Reproduction steps

- Have ServiceInsight 2.6.0 installed and configured to connect to an instance of ServiceControl
- User A connects to the machine via remote desktop and runs ServiceInsight
- ServiceInsight behavior is good/normal
- User B connects to the same machine via remote desktop while User A is still running ServiceInsight and User B now also runs ServiceInsight
- User B’s ServiceInsight will crash instantly or after clicking on endpoint(s) in endpoint explorer
- Users can close the app, and User B can run the app first
- User B will see normal behavior
- User A runs ServiceInsight with User B still using it
- Now User A will crash instantly or after clicking endpoint(s) in endpoint explorer

This was likely introduced in version 2.4 with [this issue](https://github.com/Particular/ServiceInsight/issues/941) as it is working in ServiceInsight 2.3.

**Comment by kbaley** [comment]
One possible fix for this is to incorporate the username into the name of the mutex that's used to check if ServiceInsight is running:

https://github.com/Particular/ServiceInsight/blob/master/src/ServiceInsight/App.xaml.cs#L20

**Comment by odelljl** [comment]
We have this exact issue and are holding on version 2.3 as a result.

**Comment by HEskandari** [comment]
There are two aspects to this:

- The code uses a constant as a mutext to avoid running multi-instance. When running more than one user (via RDP), this causes the second instance to not run. 
- Even in a single user scenario, there's a named pipe communication between second -> first instance, so that we can send the information back to the instance already running. For example, user has a session open already, then clicks the 'Open in ServiceInsight' button in ServicePulse. The information should appear in the instance already running (e.g. message getting selected).

The named pipe and the mutex name (both named 'ServiceInsight') need to be changed to something that allow multiple-user to use it at the same time (e.g. include the logged-in username?)

**Comment by HEskandari** [comment]
More investigation showed that the issue is not the Mutex, as in terminal model (RDP) the mutex would have local visibility which makes it invisible to the second RDP session. The actual underlying issue is an exception that happens due to the second NamedPipe being created on the machine when launching the second instance. The second named pipe would use the same name (ServiceInsight) which would conflict with the first running instance and the application crashes. The exception in the log file was like this:

```
2021-05-27 23:07:36.971 +00:00 [Error] (ServiceInsight.Framework.ExceptionHandler) An unhandled exception occurred. Error message is "All pipe instances are busy.
".
System.IO.IOException: All pipe instances are busy.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Pipes.NamedPipeServerStream.Create(String fullPipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode, PipeOptions options, Int32 inBufferSize, Int32 outBufferSize, PipeAccessRights rights, SECURITY_ATTRIBUTES secAttrs)
   at System.IO.Pipes.NamedPipeServerStream..ctor(String pipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode, PipeOptions options, Int32 inBufferSize, Int32 outBufferSize, PipeSecurity pipeSecurity, HandleInheritability inheritability, PipeAccessRights additionalAccessRights)
   at System.IO.Pipes.NamedPipeServerStream..ctor(String pipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode)
   at ServiceInsight.Startup.StartupConfigListener.<Handle>d__1.MoveNext() in /_/src/ServiceInsight/Startup/StartupConfigListener.cs:line 43
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at ServiceInsight.Startup.StartupConfigListener.<Start>d__0.MoveNext() in /_/src/ServiceInsight/Startup/StartupConfigListener.cs:line 27
2021-05-27 23:07:37.002 +00:00 [Error] (ServiceInsight.Framework.ExceptionHandler) An unhandled exception occurred. Error message is "The calling thread must be STA, because many UI components require this.".
System.InvalidOperationException: The calling thread must be STA, because many UI components require this.
   at System.Windows.Input.InputManager..ctor()
   at System.Windows.Input.InputManager.GetCurrentInputManagerImpl()
   at System.Windows.Input.KeyboardNavigation..ctor()
   at System.Windows.FrameworkElement.FrameworkServices..ctor()
   at System.Windows.FrameworkElement.EnsureFrameworkServices()
   at System.Windows.FrameworkElement..ctor()
   at System.Windows.Controls.Control..ctor()
   at System.Windows.Window..ctor()
   at ServiceInsight.Framework.ExceptionHandler.ShowExceptionDialog(Exception ex) in /_/src/ServiceInsight/Framework/ExceptionHandler.xaml.cs:line 75
   at System.Threading.Tasks.TaskScheduler.PublishUnobservedTaskException(Object sender, UnobservedTaskExceptionEventArgs ueea)
   at System.Threading.Tasks.TaskExceptionHolder.Finalize()
```

**Comment by SeanFeldman** [comment]
The issue has been fixed and released in version 2.8.
Please see the announcement [here](https://discuss.particular.net/t/serviceinsight-2-8-minor-release-available/2434).
